server/services/userService.ts

```diff
@@ -18,21 +19,23 @@ export default class UserService {
     private readonly prisonerService: PrisonerService,
   ) {}
 
-  async getUser(token: string): Promise<UserDetails> {
-    const user = await this.manageUsersApiClient.getUser(token)
-    const userCaseloads = await this.prisonerService.getUsersCaseloads(token)
+  async getUser(userToken: string): Promise<UserDetails> {
+    const user = await this.manageUsersApiClient.getUser(userToken)
+    const userCaseloads = await this.prisonerService.getUsersCaseloads(userToken)
+    const roles = this.getUserRoles(userToken)
     return {
       ...user,
-      roles: this.getUserRoles(token),
+      roles,
       ...(user.name && { displayName: convertToTitleCase(user.name) }),
       caseloads: userCaseloads.map(uc => uc.caseLoadId),
       caseloadDescriptions: userCaseloads.map(uc => uc.description),
       caseloadMap: new Map(userCaseloads.map(uc => [uc.caseLoadId, uc.description])),
+      isSupportUser: roles.includes('COURTCASE_RELEASEDATE_SUPPORT'),
```

Comment:
I'm not sure how you'll be allocating this role, could a user have this role and `ADJUSTMENTS_MAINTAINER` and then this check would stop them from being able to write adjustments? Does this check need to also check the user doesn't have `ADJUSTMENTS_MAINTAINER` role?

Reply:
Yea I thought about this. I thought it would be easier to ensure the support user has `ADJUSTMENTS_MAINTAINER` AND  `COURTCASE_RELEASEDATE_SUPPORT`. Otherwise we'd have to replace everywhere in CCRD that checks the `ADJUSTMENTS_MAINTAINER` role for access to adjustments.